### ISSUE TITLE
False positive on await for AbstractEventLoop.create_connection (regression in 0.700)

### ISSUE THREAD

**Issue by bmerry** [python/mypy#6909]
* Are you reporting a bug, or opening a feature request?

Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
import asyncio

async def foo(protocol: asyncio.StreamReaderProtocol, loop: asyncio.AbstractEventLoop):
    await loop.create_connection(lambda: protocol, 'localhost', 1234)
```

* What is the actual behavior/output?

```
await.py:4: error: Incompatible types in "await" (actual type "Generator[Any, None, Tuple[BaseTransport, BaseProtocol]]", expected type "Awaitable[Any]")
```

* What is the behavior/output you expect?

No error.

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

0.701. This appears to be a regression between 0.670 (which has no output) and 0.700. I get the error still with mypy master (0.710+dev.95a04913df54d5b1ff78fcb9e7467b798491ec4b).

* What are the mypy flags you are using? (For example --strict-optional)

No flags

**Comment by bmerry** [comment]
I also get a similar error from `AbstractEventLoop.create_server`, but not from all asyncio coroutines e.g. `Event.wait` is fine.

**Comment by ilevkivskyi** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/6802. Could you please double-check that PR https://github.com/python/mypy/pull/6821 fixes your issue as well?

**Comment by bmerry** [comment]
Thanks, I can confirm that PR #6821 fixes the issue.
